packages/sync/src/modules/class-full-sync-immediately.php

```diff
@@ -0,0 +1,395 @@
+	 *
+	 * @access public
+	 *
+	 * @param array $values New values to set.
+	 *
+	 * @return bool True if success.
+	 */
+	public function update_status( $values ) {
+		return $this->set_status( wp_parse_args( $values, $this->get_status() ) );
+	}
+
+	/**
+	 * Retrieve the status of the current full sync.
+	 *
+	 * @param array $values New values to set.
+	 *
+	 * @access public
+	 *
+	 * @return boolean Full sync status.
+	 */
+	public function set_status( $values ) {
+		return \Jetpack_Options::update_raw_option( self::STATUS_OPTION, $values );
+	}
+
+	/**
+	 * Given an initial Full Sync configuration set the initial status.
+	 *
+	 * @param array $full_sync_config Full sync configuration.
+	 *
+	 * @return array Initial Sent status.
+	 */
+	public function get_initial_progress( $full_sync_config ) {
+		// Set default configuration, calculate totals, and save configuration if totals > 0.
+		$status = array();
+		foreach ( $full_sync_config as $name => $config ) {
+			$module          = Modules::get_module( $name );
+			$status[ $name ] = array(
+				'total'    => $module->total( $config ),
+				'sent'     => 0,
+				'finished' => false,
+			);
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Get the range for content (posts and comments) to sync.
+	 *
+	 * @access private
+	 *
+	 * @return array Array of range (min ID, max ID, total items) for all content types.
+	 */
+	private function get_content_range() {
+		$range  = array();
+		$config = $this->get_status()['config'];
+		// Only when we are sending the whole range do we want to send also the range.
+		if ( true === isset( $config['posts'] ) && $config['posts'] ) {
+			$range['posts'] = $this->get_range( 'posts' );
+		}
+
+		if ( true === isset( $config['comments'] ) && $config['comments'] ) {
+			$range['comments'] = $this->get_range( 'comments' );
+		}
+
+		return $range;
+	}
+
+	/**
+	 * Get the range (min ID, max ID and total items) of items to sync.
+	 *
+	 * @access public
+	 *
+	 * @param string $type Type of sync item to get the range for.
+	 *
+	 * @return array Array of min ID, max ID and total items in the range.
+	 */
+	public function get_range( $type ) {
+		global $wpdb;
+		if ( ! in_array( $type, array( 'comments', 'posts' ), true ) ) {
+			return array();
+		}
+
+		switch ( $type ) {
+			case 'posts':
+				$table     = $wpdb->posts;
+				$id        = 'ID';
+				$where_sql = Settings::get_blacklisted_post_types_sql();
+
+				break;
+			case 'comments':
+				$table     = $wpdb->comments;
+				$id        = 'comment_ID';
+				$where_sql = Settings::get_comments_filter_sql();
+				break;
+		}
+
+		// TODO: Call $wpdb->prepare on the following query.
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$results = $wpdb->get_results( "SELECT MAX({$id}) as max, MIN({$id}) as min, COUNT({$id}) as count FROM {$table} WHERE {$where_sql}" );
+		if ( isset( $results[0] ) ) {
+			return $results[0];
+		}
+
+		return array();
+	}
+
+	/**
+	 * Continue sending.
```

Comment:
Maybe this could use some more clarity on what we're doing here.

```suggestion
	 * Continue sending instead of enqueueing.
```